### PR TITLE
chore!: drop support for Python 3.8

### PR DESCRIPTION
**Pull request by kevinzwang**
BREAKING CHANGE: Python 3.8 has reached EOL, set our minimum version to 3.9

### CODE REVIEW

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -372,7 +372,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ['3.8'] # can't use 3.7 due to requiring anon mode for adlfs
+        python-version: ['3.9'] # can't use 3.7 due to requiring anon mode for adlfs
```
Comment (samster25) [inline .github/workflows/python-package.yml:375]: let's delete these old comments as well

File: requirements-dev.txt [hunk 2]
```diff
@@ -26,73 +26,67 @@ pytest-codspeed==2.2.1
...
-pyiceberg==0.7.0; python_version >= '3.8'
-tenacity==8.2.3; python_version >= '3.8'
+pyiceberg==0.7.0
+tenacity==8.2.3
 
 # Delta Lake
 deltalake==0.5.8; platform_system == "Windows"
-deltalake==0.18.2; platform_system != "Windows" and python_version < '3.9'
-deltalake==0.19.2; platform_system != "Windows" and python_version >= '3.9'
+deltalake==0.18.2; platform_system != "Windows" and python_version < '3.10'
```
Comment (samster25) [inline requirements-dev.txt:52]: We shouldn't need to update these right?

**Review by samster25 — APPROVED** [review]
Looks good! @kevinzwang we should also bump up the minimum pyarrow pin to `8` as well!

File: requirements-dev.txt (hunk 2 above)
Comment (kevinzwang) [reply requirements-dev.txt:52]: I made these changes due to some dependency conflict before, maybe it works lemme check

**Pull request merged**